Makefile

```diff
@@ -85,7 +85,7 @@ verify:
 test:
 	go test --race --v ./pkg/...
 
-images: image-karmada-controller-manager image-karmada-scheduler image-karmada-webhook image-karmada-agent
+images: clean image-karmada-controller-manager image-karmada-scheduler image-karmada-webhook image-karmada-agent
```

Comment:
it's kind of weird. Seems we don't need to clean the binaries every time, since if there is a source code change, the binaries will be re-build.

Reply:
I built karmada on MacOS, so `GOOS ?= $(shell go env GOOS)` is darwin, then I change GOOS to `linux`,and run make,but the binary is not rebuild.

Reply:
Why do you need to change GOOS? you build the binary on `MacOS` and using it on `Linux` OS?

Reply:
yes,cross-compiling on MacOS

Reply:
I've never seen things like this(take `clean` as a dependency). This will force re-build all images every time, I'm afraid this will violate most of the case.

@daixiang0 any ideas for better support cross-compile?

Reply:
Generate multi-target based on arch in Makefile, for both binary and image.

Reply:
Are there any examples?

Reply:
I think kubeedge does.

Reply:
Is there a chance to run it on MacOS? How about just set GOOS to linux ?

Reply:
Yes, I know a lot of guys are trying karmada with Mac.

Reply:
Just found a tricky solution at [stackoverflow](https://stackoverflow.com/questions/21826550/make-redo-some-targets-if-configuration-changes).

Reply:
as discussed offline, just add bin to gitignore, need find a better way for cross-compiling